eole/modules/multi_headed_attn.py

```diff
@@ -267,7 +267,15 @@ def __init__(
         running_config=None,
         is_decoder: bool = True,
     ) -> None:
-        self.dim_per_head = model_config.hidden_size // model_config.heads
+        if hasattr(model_config, "head_dim"):
```

Comment:
@francoishernandez is there a better way to handle this in config/models.py ?

Reply:
We can probably create some @property/@computed_field (like [here](https://github.com/eole-nlp/eole/blob/9c8cc5bf0b3b7ff13f80032b583b020c400ccc9d/eole/config/inference.py#L179-L187) for instance).

Something like:
```python
@computed_field
@property
def dim_per_head(self) -> int:
    if self.head_dim is not None:
       return (
                model_config.head_dim
                if model_config.head_dim is not None
                else model_config.hidden_size // model_config.heads
            )
    else:
        return model_config.hidden_size // model_config.heads
```

Reply:
actually
```
    @computed_field
    @property
    def dim_per_head(self) -> int:
        if self.head_dim is not None:
           return self.head_dim
        else:
            return self.hidden_size // self.heads
```